### PR TITLE
Bugfix/venue selector visibility

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 
- Hide the venue selector when having directions open 

# How 
- Refactor the `BottomSheet` and `Sidebar` components to keep track of the state of the directions page 
- Call the `onDirectionsOpened` and `onDirectionsClosed` when navigating to and from the directions page
- Refactor the implementation of toggling the visibility of the floor selector to be more generic, as we need to show/hide more than one element on the map
- Refactor the `MapsIndoorsMap` component to keep track of directions and hide or show the floor selector and the venue selector

### CODE REVIEW

File: packages/map-template/src/components/BottomSheet/BottomSheet.jsx [hunk 1]
```diff
@@ -21,12 +21,12 @@ const BOTTOM_SHEETS = {
...
  * @param {function} props.onLocationsFiltered - The list of locations after filtering through the categories.
- * @param {function} props.onHideFloorSelector - Trigger the visibility of the floor selector to be hidden.
- * @param {function} props.onShowFloorSelector - Trigger the visibility of the floor selector to be shown.
  * @param {function} props.onDisableLocations - Restrict the user from interacting with the locations when in directions mode.
  * @param {function} props.onEnableLocations - Allow the user to interact with the locations when outside of directions mode.
+ * @param {function} props.onDirectionsOpened - Check if the directions page state is open.
+ * @param {function} props.onDirectionsClosed - Check if the directions page state is closed.
  */
-function BottomSheet({ currentLocation, setCurrentLocation, currentCategories, onLocationsFiltered, onHideFloorSelector, onShowFloorSelector, onDisableLocations, onEnableLocations}) {
+function BottomSheet({ currentLocation, setCurrentLocation, currentCategories, onLocationsFiltered, onDisableLocations, onEnableLocations, onDirectionsOpened, onDirectionsClosed}) {
```
Comment (ammapspeople) [inline packages/map-template/src/components/BottomSheet/BottomSheet.jsx:29]: Do we need both the pairs for locations and directionsOpened? Can they be combined into one?

Comment (andreeaceachir142) [reply packages/map-template/src/components/BottomSheet/BottomSheet.jsx:29]: For the disable/enable locations?

Comment (ammapspeople) [reply packages/map-template/src/components/BottomSheet/BottomSheet.jsx:29]: I was wondering if we need all four:

`onDisableLocations, onEnableLocations, onDirectionsOpened, onDirectionsClosed}`

Are they (pairwise) handling the same thing? Not sure.

Comment (andreeaceachir142) [reply packages/map-template/src/components/BottomSheet/BottomSheet.jsx]: That is true actually, the enable/disable location is something that is dependent on the state of directions, so I refactored the code and removed the unnecessary parameters.

**Review by ammapspeople — APPROVED** [review]
👏

**Pull request merged**